app/services/fact_importer.rb

```diff
@@ -4,22 +4,28 @@ class FactImporter
   delegate :logger, :to => :Rails
   attr_reader :counters
 
+  def deprecation_warning
+    Foreman::Deprecation.deprecation_warning('2.0', 'Use Importer Registry to register or obtain facts for a plugin')
```

Comment:
Please use a specific deprecation for each method indicating what method should replace it. Also, the deprecation warning should be for 2.2, not 2.0.